code/modules/mob/living/emote.dm

```diff
@@ -414,11 +414,60 @@
 	key_third_person = "wsmiles"
 	message = "smiles weakly."
 
+/// The base chance for your yawn to propagate to someone else if they're on the same tile as you
+#define YAWN_PROPAGATE_CHANCE_BASE 60
+/// The base chance for your yawn to propagate to someone else if they're on the same tile as you
+#define YAWN_PROPAGATE_CHANCE_DECAY 10
+
 /datum/emote/living/yawn
 	key = "yawn"
 	key_third_person = "yawns"
 	message = "yawns."
 	emote_type = EMOTE_AUDIBLE
+	cooldown = 3 SECONDS
+
+/datum/emote/living/yawn/run_emote(mob/user, params, type_override, intentional)
+	. = ..()
+	if(!. || !isliving(user))
+		return
+
+	if(!HAS_TRAIT(user, TRAIT_YAWNED_RECENTLY))
+		addtimer(CALLBACK(src, .proc/reset_yawn, user), cooldown * 3)
```

Comment:
isn't there a cooldowns system specifically for this?

Reply:
Oh yeah that's right duh

Reply:
Wait no, cause emote datums are singleton, and it seems silly to add a whole new var to /mob/living just to track cooldowns for propagated yawns.